pkg/controller/virtualmachinevolumeexport/virtualmachinevolumeexport_controller.go

```diff
@@ -0,0 +1,158 @@
+	defer func() {
+		klog.Infof("End sync VirtualMachineVolumeExport %s", request.NamespacedName)
+	}()
+
+	cachedVmvExport := &hc.VirtualMachineVolumeExport{}
+	if err := r.client.Get(context.TODO(), request.NamespacedName, cachedVmvExport); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil // Deleted VMI. Return and don't requeue.
+		}
+		return reconcile.Result{}, err
+	}
+	r.vmvExport = cachedVmvExport.DeepCopy()
+
+	syncAll := func() error {
+		// check if virtual machine volume to export is available
+		if err := r.validateVirtualMachineVolume(); err != nil {
+			return err
+		}
+		// if there is no pvc, update state to creating and create pvc
+		if err := r.syncExportPvc(); err != nil {
+			return err
+		}
+		// if pvc export is not completed, create exporter pod if it not exist
+		// if there is an exporter pod and state is complete, update completed to yes and delete it
+		if err := r.syncExporterPod(); err != nil {
+			return err
+		}
+		// if destination is local, create local pod and update readytouse to true if it not exist
+		if destination := r.getDestination(); destination == ExporterDestinationLocal {
+			if err := r.syncLocalPod(); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+	if err := syncAll(); err != nil {
+		if err2 := r.updateStateWithReadyToUse(hc.VirtualMachineVolumeExportStateError, corev1.ConditionFalse, "VmvExportIsInError", err.Error()); err2 != nil {
+			return reconcile.Result{}, err2
+		}
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileVirtualMachineVolumeExport) getDestination() string {
+	var destination string
+	if r.vmvExport.Spec.Destination.Local != nil {
+		destination = ExporterDestinationLocal
+	}
+	return destination
+}
+
+// updateStateWithReadyToUse updates conditions and state. Other Status fields are not affected. vmvExport must be DeepCopy to avoid polluting the cache.
+func (r *ReconcileVirtualMachineVolumeExport) updateStateWithReadyToUse(state hc.VirtualMachineVolumeExportState, readyToUseStatus corev1.ConditionStatus,
+	reason, message string) error {
+	r.vmvExport.Status.Conditions = util.SetConditionByType(r.vmvExport.Status.Conditions, hc.VirtualMachineVolumeExportConditionReadyToUse, readyToUseStatus, reason, message)
+	r.vmvExport.Status.State = state
+	return r.client.Status().Update(context.TODO(), r.vmvExport)
+}
+
+func (r *ReconcileVirtualMachineVolumeExport) validateVirtualMachineVolume() error {
+	klog.Infof("Validate vmv for vmvExport %s", r.vmvExport.Name)
+	// check if vmv is exist
+	vmVolume := &hc.VirtualMachineVolume{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: r.vmvExport.Namespace, Name: r.vmvExport.Spec.VirtualMachineVolume.Name}, vmVolume)
+	if err != nil {
+		return goerrors.New("fail to get virtual machine volume")
+	}
+	// check if vmv is available
+	if vmVolume.Status.State != hc.VirtualMachineVolumeStateAvailable {
```

Comment:
I guess this part can be updated later to check vmv condition array

Reply:
please add to handle this. @young

Reply:
If #14 merged, I will update to check vmv condition array. @woohhan

Reply:
we need an issue for tracking

Reply:
#19 is related.